SW-78 added migration for s series bug with mount_manager

JFSolution opened the pull request:
fix https://github.com/mrbeam/BeamOS/issues/62

Commit:
added migration for s series bug with mount_manager
https://github.com/mrbeam/BeamOS/issues/62

Commit:
added migration for s series bug with mount_manager
https://github.com/mrbeam/BeamOS/issues/62

amrsoll reviewed (changes requested):
As I explained in https://github.com/mrbeam/BeamOS/issues/62#issuecomment-869418704 , this will Not have the intended outcome because the affected devices will not be able to perform the migration.

amrsoll commented on octoprint_mrbeam/migrate.py:
```diff
@@ -214,6 +215,19 @@ def run(self):
                     self.hostname_helper_scripts()
 
+                beamos_tier, beamos_date = self.plugin._device_info.get_beamos_version()
+                if beamos_date <= datetime.strptime(
+                    "2021-06-25", "%Y-%m-%d"
+                ).date() and (
+                    self.version_previous is None
```
Also this might raise a None-comparison error if `beamos_date` is None.

Always assume that some info read from a "config" file can be None

JFSolution replied:
the get_beamos_version returns the legacy date if it can't get the configdata, but i added a condition to handle it

Commit:
changed the condition for migration to exclude the legacy image
https://github.com/mrbeam/MrBeamPlugin/pull/1284

Commit:
changed the condition for migration to exclude the legacy image
https://github.com/mrbeam/MrBeamPlugin/pull/1284

Commit:
changed condition to handle beamos_date is None
https://github.com/mrbeam/MrBeamPlugin/pull/1284

Commit:
changed condition to handle beamos_date is None
https://github.com/mrbeam/MrBeamPlugin/pull/1284

JFSolution commented:
I spoke with @Gallore and he said, we should make this migration, so all devices have the same base, and we don't have to remember this special case.
The devices will be able to migrate because the users already received a usb-stick that hotfixes this issue

khaledsherkawi committed:
Environment selection added (#1288)

* Environment selection added

* constants fixed

khaledsherkawi committed:
v0.9.7

khaledsherkawi committed:
version bump:v0.9.7.1

JFSolution committed:
SW-38 collects pressure and rpm data of airfilter in analytics (#1294)

* Collects pressure and rpm data of airfilter in analytics
* Version bump of analytics header

khaledsherkawi committed:
Design store notification added (#1289)

* last uploaded date setting added

* design store link fixed

* last uploaded date updated on design store click

* Update octoprint_mrbeam/static/js/design_store.js

Co-authored-by: Andy Werner <Gallore@users.noreply.github.com>

Co-authored-by: Andy Werner <Gallore@users.noreply.github.com>

amrsoll committed:
SW-136 Vector engraving getting skipped when min_intensity greater than 0 (#1290)

* Fix a non-sanitised input in the laser settings.
* Introduce ``observableInt()`` which forces a shadow ko.observable to be an integer.
* Parse default values too.

JFSolution committed:
Revert "SW-38 collects pressure and rpm data of airfilter in analytics (#1294)"

This reverts commit 50c15a8e81c691d8709e7fc98d44a4f4de42e2ca.

Gallore committed:
Auto-logout keeps happening. Increased timeout and added analytics calls to the trigger blacklist (#1298)

andywerner commented on octoprint_mrbeam/migrate.py:
```diff
@@ -627,6 +645,21 @@ def disable_gcode_auto_deletion(self):
+            __package_path__, self.MIGRATE_FILES_FOLDER, "fix_s_series_mount_manager"
+        )
+        dst = "/etc/rc.local"
+        exec_cmd("sudo cp {src} {dst}".format(src=src, dst=dst))
```
get the return value (Boolean) and add it to the following line (logging), so that the logged information is always correct.

amrsoll replied:
Or have it as an if statement
```python
if(exec_cmd( ... )):
  log("Great")
else:
  log("not so great")

amrsoll replied:
In fact, `exec_cmd()` already logs an error when the exit code is not `0` so I don't think this is necessary

JFSolution replied:
I did it with the if else

andywerner reviewed (approved):
approved with minor change requests.

khaledsherkawi committed:
Design store notify new designs (#1296)

* last uploaded date setting added

* design store link fixed

* last uploaded date updated on design store click

* Update octoprint_mrbeam/static/js/design_store.js

Co-authored-by: Andy Werner <Gallore@users.noreply.github.com>

* styling fix for switching between design store environments

* don't show notification if uploaded value is undefined in users.yaml

Co-authored-by: Andy Werner <Gallore@users.noreply.github.com>

Gallore committed:
SW-171: Trying to solve the constant re-login issue

* OK, next try: Don't do any passive logins during boot phase (and 10 secs after), also wait 3 secs before actually performing the passive login.

* fixed logging issue

* return condition needs to be the other way round

Commit:
SW-78 change to run the migration only once

Commit:
Merge remote-tracking branch 'origin/b_mount_manager_s_series' into SW-78-migrate-fix-for-mount-manager

# Conflicts:
#	octoprint_mrbeam/__version.py
#	octoprint_mrbeam/migrate.py

JFSolution committed:
SW-207 fix migration for s series (#1303)

* added migration for s series bug with mount_manager
https://github.com/mrbeam/BeamOS/issues/62

* changed the condition for migration to exclude the legacy image
https://github.com/mrbeam/MrBeamPlugin/pull/1284

* changed condition to handle beamos_date is None
https://github.com/mrbeam/MrBeamPlugin/pull/1284

* SW-207 fix migration won't run for s-series images

* SW-207 fix migration won't run for s-series images

Commit:
SW-207 refactor beamos_date

Commit:
SW-78 refactor beamos_date

Commit:
Merge remote-tracking branch 'origin/b_mount_manager_s_series' into SW-78-migrate-fix-for-mount-manager

JFSolution commented:
merged changes to fix that the migration won't start for s series images

JFSolution commented:
Tested it on my device, the migration works now

JFSolution commented:
fixed the branch with @amrsoll new PR https://github.com/mrbeam/MrBeamPlugin/pull/1307
